### PR TITLE
Fix symbolic bug

### PR DESCRIPTION
**Pull request by mjohnson541**
Fix bug when feeding symbolic variables through getkfkrev.

### CODE REVIEW

**Comment by mjohnson541** [comment]
I've also added finite differences as a Jacobian source and improved our handling of jac_prototype.

**Comment by mjohnson541** [comment]
Decided to not add finite differences because how much the accuracy of the Jacobian degrades.

File: src/PhaseState.jl [hunk 1]
```diff
@@ -181,12 +182,12 @@ export getKcs
 Calculates the forward and reverse rate coefficients for a given reaction, phase and state
 Maintains diffusion limitations if the phase has diffusionlimited=true
 """
-@inline function getkfkrev(rxn::ElementaryReaction,ph::U,T::W1,P::W2,C::W3,N::W4,ns::Q1,Gs::Q2,diffs::Q3,V::W5,phi::W8;kf::W6=-1.0,f::W7=-1.0) where {U<:AbstractPhase,W8,W6,W7,W5,W4,W1,W2,W3<:Real,Q1,Q2,Q3<:AbstractArray}
+@inline function getkfkrev(rxn::ElementaryReaction,ph::U,T::W1,P::W2,C::W3,N::W4,ns::Vector{Q1},Gs::Q2,diffs,V::W5,phi::W8;kf::W6=-1.0,f=-1.0) where {U<:AbstractPhase,W8,W6,W5,W4,W1,W2,W3,Q1<:Real,Q2}
```
Comment (hwpang) [inline src/PhaseState.jl:185]: Why do we remove typing on `diffs`?

File: src/PhaseState.jl [hunk 2]
```diff
@@ -181,12 +182,12 @@ export getKcs
...
     if signbit(kf) 
-        if signbit(f)
-            kf = getkf(rxn,ph,T,P,C,ns,V,phi)
-        else
+        if isa(f,Num) || !signbit(f)
```
Comment (hwpang) [inline src/PhaseState.jl:187]: Can you run a formatting script over your changes or this file?

File: src/Reactor.jl [hunk 3]
```diff
@@ -310,7 +327,7 @@ function Reactor(domain::T, y0unlumped::Array{W1,1}, tspan::Tuple, reducedmodelm
     if modelingtoolkit
         sys = modelingtoolkitize(ode)
         jac = eval(ModelingToolkit.generate_jacobian(sys)[2])
-        odefcn = ODEFunction(dydt; jac=jac, paramjac=jacp!)
+        odefcn = ODEFunction(dydt; jac=jac, paramjac=jacp!, jac_prototype=jac_prototype)
```
Comment (hwpang) [inline src/Reactor.jl:330]: Is `modelingtoolkitize` tested in our test? Maybe we should add a test

**Review by hwpang — CHANGES_REQUESTED** [review]
Thanks for the PR! I have some questions and requests for improving code formatting.

File: src/Reactor.jl [hunk 4]
```diff
@@ -90,7 +96,7 @@ function Reactor(domain::T, y0::Array{T1,1}, tspan::Tuple, interfaces::Z=[]; p::
     end
     return Reactor(domain, interfaces, y0, tspan, p, ode, recsolver, forwardsensitivities, forwarddiff, modelingtoolkit, tau, precsundials, psetupsundials, precsjulia)
 end
-function Reactor(domains::T, y0s::W1, tspan::W2, interfaces::Z=Tuple(), ps::X=SciMLBase.NullParameters(); forwardsensitivities=false, modelingtoolkit=false, tau=1e-3) where {T<:Tuple,W1<:Tuple,Z,X,W2}
+function Reactor(domains::T, y0s::W1, tspan::W2, interfaces::Z=Tuple(), ps::X=SciMLBase.NullParameters(); forwardsensitivities=false, forwarddiff=false, modelingtoolkit=false, tau=1e-3) where {T<:Tuple,W1<:Tuple,Z,X,W2}
```
Comment (hwpang) [inline src/Reactor.jl:99]:
```suggestion
function Reactor(domains::T, y0s::W1, tspan::W2, interfaces::Z=Tuple(), ps::X=SciMLBase.NullParameters(); forwardsensitivities=false, modelingtoolkit=false, tau=1e-3) where {T<:Tuple,W1<:Tuple,Z,X,W2}
```

File: src/PhaseState.jl (hunk 1 above)
Comment (mjohnson541) [reply src/PhaseState.jl:185]: It was an unnecessary definition, it doesn't optimize or help us dispatch.

File: src/Reactor.jl (hunk 3 above)
Comment (mjohnson541) [reply src/Reactor.jl:330]: We should, but we need a very small mechanism to make it fast.

File: src/PhaseState.jl (hunk 2 above)
Comment (mjohnson541) [reply src/PhaseState.jl:187]: I'm having difficulty getting the formatter extension to connect with the julia kernel. Was there anything in particular you had in mind?

Comment (hwpang) [reply src/PhaseState.jl:187]: Could you add blank after comma?

Comment (hwpang) [reply src/PhaseState.jl]: I ran formatter on the files changed in this PR, so I'm good with the formatting

File: src/PhaseState.jl (hunk 1 above)
Comment (hwpang) [reply src/PhaseState.jl]: Should remove other ones that are unnecessary too?

**Comment by mjohnson541** [comment]
Is this good to merge?

File: .github/workflows/Install.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    branches:
+      - main
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: julia-actions/setup-julia@v1
+        with:
+          version: 1.8
```
Comment (hwpang) [inline .github/workflows/Install.yml:16]: Why do we pin to 1.8? What is preventing us to move up?

**Review by hwpang — COMMENTED** [review]
One last question, otherwise LGTM

**Pull request closed without merging**